src/util/upnp_clients.cc

```diff
@@ -146,12 +147,16 @@ void Clients::addClientByDiscovery(const struct sockaddr_storage* addr, const st
 #endif
 }
 
-void Clients::getInfo(const struct sockaddr_storage* addr, const std::string& userAgent, const ClientInfo** ppInfo)
+void Clients::getInfo(std::shared_ptr<Config> config, const struct sockaddr_storage* addr, const std::string& userAgent, const ClientInfo** ppInfo)
```

Comment:
instead of injecting here the config:

Add an additional function `addClienInfo(ClientInfo* add)` to manipulate `ClientInfo clientInfo[]`.
Probably in server.cc get the configuration and call this function.